odxtools/nameditemlist.py

```diff
@@ -55,57 +53,93 @@ def append(self, item: T):
         if iskeyword(item_name):
             item_name = f"{item_name}_"
 
+        # eliminate conflicts between the item name and existing
+        # attributes of the NamedItemList object
         i = 1
         tmp = item_name
         while True:
-            if tmp not in self.__dict__:
-                self.__dict__[tmp] = item
-                self._typed_dict[tmp] = item
-                return tmp
+            # using dir() for checking if there is a name conflict
+            # might be slow, but NamedItemList is not meant for
+            # ginormous lists...
+            if tmp not in dir(self):
+                break
 
             i += 1
             if item_name.endswith("_"):
                 tmp = f"{item_name}{i}"
             else:
                 tmp = f"{item_name}_{i}"
+        item_name = tmp
+
+        self.__dict__[item_name] = item
+        self._names.append(item_name)
+        self._values.append(item)
+
+    def sort(self, key: Optional[Callable[[T], str]] = None, reverse: bool = False) -> None:
```

Comment:
to sort the list, a la

```python
foo = NamedItemList(lambda x: f"item_{x}", [5, 1, 7])
print(foo)
# [item_5, item_1, item_7]
foo.sort()
# [item_1, item_5, item_7]
```

Reply:
I mean why would the user of library need to change the order of the NamedItemList ?

There is a difference between getting a sorted list (using `sorted` function), and changing the order inside the list itself

Reply:
> I mean why would the user of library need to change the order of the NamedItemList ?

not sure. it might come in handy at some point?

> There is a difference between getting a sorted list (using sorted function), and changing the order inside the list itself

yes: using `sorted()` makes a normal list, i.e., the ability to access items by name gets lost. `.sort()` gets around this. That said,
```python
foo = NamedItemList(lambda x: x, ["a", "c", "b"])
foo.sort()
```
and
```python
foo = NamedItemList(lambda x: x, ["a", "c", "b"])
foo = NamedItemList(lambda x: x, sorted(foo))
```

should be equivalent...